src/iminuit/qtwidget.py

```diff
@@ -0,0 +1,413 @@
+            self.tmin = QtWidgets.QDoubleSpinBox(
+                alignment=QtCore.Qt.AlignmentFlag.AlignCenter
+            )
+            self.tmin.setRange(_make_finite(-np.inf), _make_finite(np.inf))
+            self.tmax = QtWidgets.QDoubleSpinBox(
+                alignment=QtCore.Qt.AlignmentFlag.AlignCenter
+            )
+            self.tmax.setRange(_make_finite(-np.inf), _make_finite(np.inf))
+            self.tmin.setSizePolicy(size_policy)
+            self.tmax.setSizePolicy(size_policy)
+            self.fix = QtWidgets.QPushButton("Fix")
+            self.fix.setCheckable(True)
+            self.fix.setChecked(minuit.fixed[par])
+            self.fit = QtWidgets.QPushButton("Fit")
+            self.fit.setCheckable(True)
+            self.fit.setChecked(False)
+            size_policy = QtWidgets.QSizePolicy(
+                QtWidgets.QSizePolicy.Policy.Fixed, QtWidgets.QSizePolicy.Policy.Fixed
+            )
+            self.fix.setSizePolicy(size_policy)
+            self.fit.setSizePolicy(size_policy)
+            layout1 = QtWidgets.QHBoxLayout()
+            layout.addLayout(layout1)
+            layout1.addWidget(label)
+            layout1.addWidget(self.slider)
+            layout1.addWidget(self.value_label)
+            layout1.addWidget(self.fix)
+            layout2 = QtWidgets.QHBoxLayout()
+            layout.addLayout(layout2)
+            layout2.addWidget(self.tmin)
+            layout2.addWidget(self.tmax)
+            layout2.addWidget(self.fit)
+
+            self.reset(minuit.values[par], limits=minuit.limits[par])
+
+            self.tmin.setSingleStep(1e-1 * (self.vmax - self.vmin))
+            self.tmax.setSingleStep(1e-1 * (self.vmax - self.vmin))
+            self.tmin.setMinimum(_make_finite(minuit.limits[par][0]))
+            self.tmax.setMaximum(_make_finite(minuit.limits[par][1]))
+
+            self.slider.valueChanged.connect(self.on_val_changed)
+            self.fix.clicked.connect(self.on_fix_toggled)
+            self.tmin.valueChanged.connect(self.on_min_changed)
+            self.tmax.valueChanged.connect(self.on_max_changed)
+            self.fit.clicked.connect(self.on_fit_toggled)
+
+        def _int_to_float(self, value):
+            return self.vmin + (value / 1e8) * (self.vmax - self.vmin)
+
+        def _float_to_int(self, value):
+            return int((value - self.vmin) / (self.vmax - self.vmin) * 1e8)
+
+        def on_val_changed(self, val):
+            val = self._int_to_float(val)
+            self.value_label.setText(f"{val:.3g}")
+            minuit.values[self.par] = val
+            self.callback()
+
+        def on_min_changed(self):
+            tmin = self.tmin.value()
+            if tmin >= self.vmax:
+                self.tmin.blockSignals(True)
+                self.tmin.setValue(self.vmin)
+                self.tmin.blockSignals(False)
+                return
+            self.vmin = tmin
+            self.slider.blockSignals(True)
```

Comment:
Since you use signal blocking repeatedly, it is worthwhile to add a context manager for this. It makes the code more readable, and in case of exceptions, signals are still unblocked.

```py
from contextlib import contextmanager

@contextmanger
def SignalBlocker(obj):
     obj.blockSignals(True)
     yield
     obj.blockSignals(False)


# usage

with SignalBlock(obj):
     # do stuff where signals of obj are blocked

Reply:
Makes sense; I changed it in [9485f85](https://github.com/scikit-hep/iminuit/pull/1056/commits/9485f85c203db9d276942ecd88a8a1ceff6295df)